release notes/v0.29.0.md

```diff
@@ -0,0 +1,133 @@
+
+    client.close();
+}
+```
+
+You can find more information and examples how to use k6's new gRPC testing capabilities in [our documentation](https://k6.io/docs/javascript-api/k6-net-grpc).
+
+Huge thanks to @rogchap for adding this feature!
+
+### New options for configuring DNS resolution ([#1612](https://github.com/loadimpact/k6/pull/1612))
+
+You can now control some aspects of how k6 performs DNS resolution! Previously, k6 would have cached DNS responses indefinitely ([#726](https://github.com/loadimpact/k6/issues/726)) and always picked the first resolved IP ([#738](https://github.com/loadimpact/k6/issues/738)) for all connections. This caused issues, especially when load testing services that relied on DNS for load-balancing or auto-scaling.
+
+For technical reasons explored in ([#726](https://github.com/loadimpact/k6/issues/726)), k6 v0.29.0 still doesn't respect the actual TTL value of resolved IPs, that will be fixed in a future k6 version. For now, it simply allows users to specify a global static DNS TTL value and resolution strategy manually. It also has better defaults! Now, by default, the global DNS TTL value is 5 minutes and, if the DNS resolution returned multiple IPs, k6 will pick a random (preferably IPv4) one for each connection.
+
+You can also configure this behavior with the new `--dns` CLI flag, the `K6_DNS` environment variable, or the `dns` script/JSON option. Three DNS resolution options are exposed in this k6 version: `ttl`, `select`, and `policy`.
+
+Possible `ttl` values are :
+- `0`: no caching at all - each request will trigger a new DNS lookup.
+- `inf`: cache any resolved IPs for the duration of the test run _(the old k6 behavior)_.
+- any time duration like `60s`, `5m30s`, `10m`, `2h`, etc.; if no unit is specified (e.g. `ttl=3000`), k6 assumes milliseconds. _The new default value is `5m`._
+
+Possible `select` values are:
+- `first` - always pick the first resolved IP _(the old k6 behavior)_.
+- `random` - pick a random IP for every new connection _(the new default value)_.
+- `roundRobin` - iterate sequentially over the resolved IPs.
+
+Possible `policy` values are:
+- `preferIPv4`: use IPv4 addresses, if available, otherwise fall back to IPv6 _(the new default value)_.
+- `preferIPv6`: use IPv6 addresses, if available, otherwise fall back to IPv4.
+- `onlyIPv4`: only use IPv4 addresses, ignore any IPv6 ones.
+- `onlyIPv6`: only use IPv6 addresses, ignore any IPv4 ones.
+- `any`: no preference, use all addresses _(the old k6 behavior)_.
+
+Here are some configuration examples:
+
+```bash
+k6 run --dns "ttl=inf,select=first,policy=any" script.js # this is the old k6 behavior
+K6_DNS="select=random,ttl=5m,policy=preferIPv4" k6 cloud script.js # new default behavior
+# syntax for the JSON config file or for the exported script `options`:
+echo '{"dns": {"select": "roundRobin", "ttl": "1h33m7s", "policy": "onlyIPv6"}}' > config.json
+k6 run --config "config.json" script.js
```

Comment:
Nit: this can be a one-liner:

```bash
k6 run --config <(echo '{"dns": {"select": "roundRobin", "ttl": "1h33m7s", "policy": "onlyIPv6"}}') script.js
```

But maybe it's clearer this way.

Reply:
yeah ... I don't think code golfing the examples is good idea :P

Reply:
Don't think this qualifies as "code golfing" 😄, it's a well known redirection feature, but sure, it might not be as clear.